rfcs/text/0428-extending-and-duplicating-types.md

```diff
@@ -0,0 +1,970 @@
+      "$ref": "https://blockprotocol.org/@alice/property-type/name/v/1"
+    },
+    "https://blockprotocol.org/@bob/property-type/tenure": {
+      "$ref": "https://blockprotocol.org/@bob/property-type/tenure/v/1"
+    },
+    "https://blockprotocol.org/@alice/property-type/occupation": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/occupation/v/1"
+    }
+  },
+  "required": [
+    "https://blockprotocol.org/@alice/property-type/name",
+    "https://blockprotocol.org/@bob/property-type/tenure",
+    "https://blockprotocol.org/@alice/property-type/occupation"
+  ]
+}
+```
+
+As the intermediate expanded entity type does not declare any supertypes, we are unable to trivially substitute this version of Bob's `Employee` in place of `Person`.
+
+## Detecting cycles
+
+> 💭 Because of how verisoning is specified in the [versioning RFC](./0408-versioning-types.md), entity types cannot create proper dependency cyles. The dependency cycles explained below are not about the literal type hierarchy, but rather about indirection and obfuscation of types and potential incompatabiliites that can be introduced. Cycles across type versions can still result in valid types because the types are immutable and that we are dealing with composition.
+
+An extension cycle happens when a part of an inheritance tree revisits a base URL it has already seen. As a contrived example, an entity type `Country` could be the supertype of `Region`, which in turn could be a supertype of the same `Country` entity type.
+
+First entity type version of `Country` without supertype:
+
+```json
+{
+  "$schema": "https://blockprotocol.org/types/modules/graph/0.3/schema/entity-type",
+  "kind": "entityType",
+  "$id": "https://blockprotocol.org/@alice/entity-type/country/v/1",
+  "type": "object",
+  "title": "Country",
+  "properties": {
+    "https://blockprotocol.org/@alice/property-type/name": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/name/v/1"
+    }
+  },
+  "required": ["https://blockprotocol.org/@alice/property-type/name"]
+}
+```
+
+Entity type `Region` with `Country` as a supertype
+
+```json
+{
+  "$schema": "https://blockprotocol.org/types/modules/graph/0.3/schema/entity-type",
+  "kind": "entityType",
+  "$id": "https://blockprotocol.org/@alice/entity-type/region/v/1",
+  "type": "object",
+  "title": "Region",
+  "allOf": [
+    { "$ref": "https://blockprotocol.org/@alice/entity-type/country/v/1" }
+  ],
+  "properties": {
+    "https://blockprotocol.org/@alice/property-type/blurb": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/blurb/v/1"
+    }
+  },
+  "required": ["https://blockprotocol.org/@alice/property-type/blurb"]
+}
+```
+
+The second, cyclic version of the `Country` entity type
+
+```json
+{
+  "$schema": "https://blockprotocol.org/types/modules/graph/0.3/schema/entity-type",
+  "kind": "entityType",
+  // Because of versioning, we cannot change this version to /v/1 and create a "proper" cycle.
```

Comment:
Is this true? 

The schemas are just plain records, we don't have any concept of the times they were created at. It would be possible to publish two types that depend on each other in the same version. I _could_ write a type which depends on itself in version 1.